features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsView.kt

```diff
@@ -80,33 +85,50 @@ fun RoomDetailsView(
             .padding(padding)
             .verticalScroll(rememberScrollState())
         ) {
-            HeaderSection(
-                avatarUrl = state.roomAvatarUrl,
-                roomId = state.roomId,
-                roomName = state.roomName,
-                roomAlias = state.roomAlias
-            )
-
-            ShareSection(onShareRoom = onShareRoom)
+            if (state.dmMember == null) {
+                RoomHeaderSection(
+                    avatarUrl = state.roomAvatarUrl,
+                    roomId = state.roomId,
+                    roomName = state.roomName,
+                    roomAlias = state.roomAlias
+                )
+                RoomShareSection(onShareUser = onShareRoom)
+            } else {
+                RoomMemberHeaderSection(
+                    avatarUrl = state.roomAvatarUrl ?: state.dmMember.avatarUrl,
+                    userId = state.dmMember.userId,
+                    userName = state.roomName
+                )
+                RoomMemberShareSection(onShareUser = { onShareMember(state.dmMember) })
+            }
 
             if (state.roomTopic != null) {
                 TopicSection(roomTopic = state.roomTopic)
             }
 
-            val memberCount = (state.memberCount as? Async.Success<Int>)?.state
-            MembersSection(
-                memberCount = memberCount,
-                isLoading = state.memberCount.isLoading(),
-                openRoomMemberList = openRoomMemberList
-            )
+            if (state.dmMember == null) {
```

Comment:
Introducing a sealed interface to avoid checking on nullability would probably be cleaner